src/commands/exportApi.ts

```diff
@@ -6,17 +6,25 @@ import * as fs from "fs-extra";
 import * as path from "path";
 import * as vscode from "vscode";
 import { ApiSpecExportResultFormat } from "../azure/ApiCenter/contracts";
-import { ApiCenterVersionDefinitionManagement } from "../azure/ApiCenterDefines/ApiCenterDefinition";
+import { ApiCenterVersionDefinitionDataPlane, ApiCenterVersionDefinitionManagement } from "../azure/ApiCenterDefines/ApiCenterDefinition";
+import { TreeViewType } from "../constants";
 import { ext } from "../extensionVariables";
 import { ApiVersionDefinitionTreeItem } from "../tree/ApiVersionDefinitionTreeItem";
+import { ApiServerItem } from "../tree/DataPlaneAccount";
+import { SubscriptionTreeItem } from "../tree/SubscriptionTreeItem";
+import { UiStrings } from "../uiStrings";
 import { createTemporaryFolder } from "../utils/fsUtil";
 import { GeneralUtils } from "../utils/generalUtils";
 export namespace ExportAPI {
     export async function exportApi(
         context: IActionContext,
         node?: ApiVersionDefinitionTreeItem): Promise<void> {
         if (!node) {
-            node = await ext.treeDataProvider.showTreeItemPicker<ApiVersionDefinitionTreeItem>(new RegExp(`${ApiCenterVersionDefinitionManagement.contextValue}*`), context);
+            let res = await getDefinitionTreeNode(context);
```

Comment:
Do we need `res`? Just `node = await getDefinitionTreeNode(context);` is OK?

Reply:
no, the getDefinitionTreeNode return type is node|null,  so can not assign null to node.

Reply:
node is `typeX | undefiend`, while getDefinitionTreeNode is `typeX | null`.

There is another related comment: getDefinitionTreeNode is better to return `typeX | undefiend`. I rarely see a fucntion return `typeX | null`, but more are `typeX | undefiend`.

With above impreovement, we could also remove `res`.

Reply:
okay